librosa/core/time_frequency.py

```diff
@@ -1227,8 +1234,23 @@ def fourier_tempo_frequencies(sr=22050, win_length=384, hop_length=512):
     return fft_frequencies(sr=sr * 60 / float(hop_length), n_fft=win_length)
 
 
+def _weighting_function(func):
+    @functools.wraps(func)
+    def inner(frequencies, min_db=-80.0):
+        # Vectorize to make our lives easier
+        frequencies = np.asanyarray(frequencies)
+
+        # Call weighting function (with pre-computed squared frequency)
+        weights = func(frequencies**2.0)
+        if min_db is not None:
+            weights = np.maximum(min_db, weights)
+        return weights
+    return inner
```

Comment:
I see why you wrote it this way, but it would be more accessible if we just made this a normal subroutine instead of a decorator.  In general we try to keep python fanciness to a minimum :grin:

Reply:
good thing I restrained myself from setting the docstring with the decorator too 😝

So to clarify, you want me to just duplicate those lines directly into each weighting function and just get rid of the decorator? 

I just like how the weighting functions are the equation only, but however you want it since you're the maintainer